HARDHAT_DEV_NOTES.md

```diff
@@ -0,0 +1,110 @@
+# periodicRewardsInjector
+
+## Intro
+The ChildChainGaugeInjector is designed to manage weekly injections of non-bal tokens into  Child Chain layer zero gauges based on a predefined keeper using chainlink keepers.
+It is meant to be deployed for a single ERC20 reward token, but can manage any number of gauges for that token.
+
+It allows the configured admin to define a list of schedules, and then can be operated by [Chainlink Automation](https://automation.chain.link/).
+
+Methods exists to add new schedules without implicating the already running schedules for a gauge.
+
+The contract also includes functionality for the owner to sweep ERC20 tokens and gas tokens in order to allow the contract to be easily decommissioned should there be any issues or it is no longer needed.
+
+
+### The Child Chain Gauge runs on weekly epochs:
+
+- Only the defined distributor for a given token may inject rewards.  
+- The injector uses changes to period_finish on the gauge contract to understand epochs and runs once per epoch as early as possible.
+
+This contract is intended to operate as the distributor, and has functionality to return distributorship to the owner
+
+### The watchlist
+
+The injector runs using a watch list. The watch list is defined as the tuple of [gaugeAddress, amountPerPeriod, maxPeriods, doNotStartBeforeTimestamp].
+
+For every streamer address, assuming a sufficent token balance, the injector will inject the specified amounts each epoch until it has done so maxPeriods time.
+
+It's possible to add new recipients and configurations without altering or including the old ones (difference to v1).
+
+This list is defined by calling the function `addRecipients(streamerAddresses, amountsPerPeriod, maxPeriods, doNotStartBeforeTimestamp)` on the deployed injector.
```

Comment:
here maybe i'm being picky, but instead of `streamerAddresses` seems more matching `gauges/recipients` to be matching the same language as in the contract

applicable in other instances where `streamer` had being used